features/step_definitions/icon_steps.rb

```diff
@@ -10,17 +10,35 @@
 Given /^I have an icon uploaded$/ do
   step "I am editing a pseud"
   step %{I attach the file "test/fixtures/icon.gif" to "icon"}
-    step %{I press "Update"}
+  step %{I press "Update"}
 end
 
 ### WHEN
 
-When /^I add an icon to the collection$/ do
+When /^I add an icon to the collection "([^"]*)"$/ do |name|
   step %{I am logged in as "moderator"}
-    step %{I am on "Pretty" collection's page}
-    step %{I follow "Settings"}
-    step %{I attach the file "test/fixtures/icon.gif" to "collection_icon"}
-    step %{I press "Update"}
+  step %{I am on "#{name}" collection's page}
+  step %{I follow "Settings"}
+  step %{I attach the file "test/fixtures/icon.gif" to "collection_icon"}
+  step %{I press "Update"}
+end
+
+When /^I delete the icon from the collection "([^"]*)"$/ do |name|
+  step %{I am logged in as "moderator"}
+  step %{I am on "#{name}" collection's page}
+  step %{I follow "Settings"}
+  check("collection_delete_icon")
+  step %{I press "Update"}
+end
+
+Then /^the collection "([^"]*)" has an icon$/ do |name|
```

Comment:
Since this is a `then` step and we're checking whether an expectation is met rather than performing an action, it would be better to phrase this as `the collection "([^"]*)" should have an icon`.  If it screams bloody murder about an ambiguous match due to a model-related pickle step, `the "([^"]*)" collection should have an icon` will probably work.

Also, I'd suggest looking up the collection by title rather than name, so it's in line with most of other collection-related steps. (We do `find_by_name` in 5 places, but `find_by_title` in 53.)